src/main/java/com/todo/backend/Controller.java

```diff
@@ -1,54 +1,302 @@
 package com.todo.backend;
 
-import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
+import com.todo.backend.Metrics.LastMetrics;
+import com.todo.backend.ToDo.Priority;
+
+import java.text.ParseException;
+import java.time.Instant;
 import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
 import java.util.List;
+import java.util.Map;
 
+@CrossOrigin(origins = "*")
 @RestController
 public class Controller {
 
-    private List<String> items = new ArrayList<>();
+    // #region ################################ DATA BASE
+    private List<ToDo> DB = new ArrayList<>();
+    private Metrics metrics = new Metrics();
+
+    private SearchParams previousParams = new SearchParams();
+    private List<ToDo> lastResponse = null;
+    // #endregion
+
+    // #region ################################ FILTER & SORTING
+    private ResponseEntity<String> toggleDoneUndone(int id, Integer inPage, boolean markedAs) {
```

Comment:
is the inPage working like "all todos in page selected" ? if so I would vote more for different endpoint for batch processing